src/Components/Components/src/BindConverter.cs

```diff
@@ -1366,6 +1395,10 @@ private static class FormatterDelegateCache
                     {
                         formatter = (BindFormatter<T>)FormatEnumValueCore<T>;
                     }
+                    else if (typeof(T).IsArray)
+                    {
+                        formatter = (BindFormatter<T>)FormatArrayValueCore<T>;
```

Comment:
Since this is being cached, do you think it might be worth going through reflection to get `FormatArrayValueCore<U>`, where `U` is the element type in the array?

If we did this, it seems like the logic in `FormatArrayValueCore` would be a lot simpler and could directly call `array[i].ToString(culture)` without having to box the array entry along the way.

Reply:
Actually I guess it would end up looking like this:

```cs
if (typeof(T) == typeof(int[]))
{
    formatter = (BindFormatter<T>)MakeArrayFormatter<int>((value, culture) => value.ToString(culture));
}
else if (typeof(T) == typeof(DateTime[]))
{
    formatter = (BindFormatter<T>)MakeArrayFormatter<DateTime>((value, culture) => value.ToString(culture));
}
else
...
```

... just so it's possible to reference the culture-specific `ToString` methods on each type. Overall this is pretty nasty.

I was going to suggest maybe the best solution is to strictly require people to use culture-invariant formatting for select option values, but thinking about it more, I'm not sure that's even possible. If we're relying on `Object.ToString()` as a way of stringifying the values, we can't avoid the culture-sensitive formatting, since `((object)DateTime.Now).ToString()` produces a culture-specific string.

Another way we might be able to make this sane is by limiting the range of bindable array types to things that are inherently culture-invariant when you call `ToString` on them, i.e., `string`, `int`, `long`, and enum values. I can't think of any cases where someone would want to create a multi-select dropdown in which the values are not one of these types, so it's not much of a functional loss, and we could potentially do something more complex in the future to enable culture-sensitive array types if the need arose.

What do you think? Am I making a bigger problem about this than it should be? I'm just trying to make sure we have a definite clear story about how the parsing/formatting will interact with the culture here.

Reply:
Note, I don't really know that `int` is inherently culture-invariant when you call `Object.ToString(someInt)`.

Reply:
If we really wanted to, we we could do something similar to minimal APIs and reflection look for `ToString(format, CultureInfo)` on the type and use that it's it's available. That said, I do agree, I can't think of lots of scenarios where you have multi-select with formatted values (although something like "Pick your preferred appointment times:" comes to mind.).

Reply:
I don't have an issue with limiting this to non-culture-sensitive types to start with. As you say we can always enhance this in the future if need be.

Reply:
The approach I've gone with for now is similar to the one I took in `ParserDelegateCache`. It uses reflection to call a `MakeArrayFormatter<T>` method with `T` being the element type, which in turn calls `Get<T>` to get the element formatter. The array formatter method is then returned as a `BindFormatter<T[]>`, which captures the element formatter and uses it to format the elements of the array. Aside from the initial setup/caching, this should be pretty efficient since the formatter delegate is cached and the elements of the array don't have to be boxed. Let me know if anyone sees a problem with this approach.